### PR TITLE
Create StreamConsumer and StreamMetadataProvider interfaces for splitting the consumers created by StreamConsumerFactory

### PR DESCRIPTION
**Pull request by npawar**
As part of https://github.com/linkedin/pinot/issues/2583 we plan to make the code accept any stream and not be kafka specific. 
This PR attempts to separate and abstract out the StreamConsumer and StreamMetadataProvider

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java [hunk 1]
```diff
@@ -209,10 +209,12 @@ private static String getGroupIdFromRealtimeDataTable(String realtimeTableName,
   private static class KafkaPartitionsCountFetcher implements Callable<Boolean> {
     private int _partitionCount = -1;
     private final StreamMetadata _streamMetadata;
+    private StreamConsumerFactory _streamConsumerFactory;
     private Exception _exception;
 
     private KafkaPartitionsCountFetcher(StreamMetadata streamMetadata) {
       _streamMetadata = streamMetadata;
+      _streamConsumerFactory = StreamConsumerFactoryProvider.create(_streamMetadata);
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java]: this class should be moved to a kafka specific package. pls add a TODO

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaConnectionHandler.java [hunk 2]
```diff
@@ -186,9 +198,9 @@ public ConnectingToBootstrapNode() {
     @Override
     public void process() {
       // Connect to a random bootstrap node
-      if (_simpleConsumer != null) {
+      if (_kafkaSimpleConsumer != null) {
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaConnectionHandler.java]: the package name qualifies simpleConsumer, and this is in a Kafka specific class, so renaming is not necessary

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java [hunk 3]
```diff
@@ -1348,8 +1348,9 @@ private Exception getException() {
     @Override
     public Boolean call() throws Exception {
 
+      String clientId = PinotLLCRealtimeSegmentManager.class.getSimpleName() + "_" + _topicName + "_" + _partitionId;
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java]: Is this an underscore as opposed to a dash in the PinotIdealStateBuilder as the separator?

Comment (npawar) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java]: Fixed

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java (hunk 1 above)
Comment (npawar) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java]: Added TODO for this and also for KafkaOffsetFetcher in PinotLLCRealtimeSegmentManager

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaConnectionHandler.java (hunk 2 above)
Comment (npawar) [reply pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/KafkaConnectionHandler.java]: removed unnecessary renaming

**Pull request merged**